packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -82,14 +82,46 @@ export const useStoreSwitcher = props => {
             // Do nothing when store view is not present in available stores
             if (!availableStores.get(storeCode)) return;
 
+            const previousStoreCode = storage.getItem('store_view_code');
+
             storage.setItem('store_view_code', storeCode);
             storage.setItem(
                 'store_view_currency',
                 availableStores.get(storeCode).currency
             );
 
-            // Refresh the page to re-trigger the queries once code/currency are saved in local storage.
-            history.go(0);
+            // Handle updating the URL if the store code should be present
+            if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+                const pathName = window.location.pathname;
+
+                // Check to see if we're on a page outside of the homepage
+                if (pathName !== '' && pathName !== '/') {
+                    const pathNameSplit = pathName.split('/');
+                    const pathStoreCode = pathNameSplit[1];
+
+                    // Is the current store code in the URL? If so replace it with the new one
+                    if (previousStoreCode === pathStoreCode) {
```

Comment:
I think you can use `availableStores.get(pathStoreCode).isCurrent` here

Reply:
While you could, the required code is actually much longer: `pathStoreCode && availableStores.get(pathStoreCode) && availableStores.get(pathStoreCode).isCurrent`. So I think it's best to keep the shorter version for readability.

Reply:
Okay updated it with the `has` then `get` API, thanks @eug123